cloudferrylib/os/actions/vm_snapshot.py

```diff
@@ -48,16 +48,55 @@ def run(self, *args, **kwargs):
         state_to_record = {}
         for vm in self.get_list_of_vms():
             state_to_record.update({vm.id: vm.status})
-        return {self.namespace_variable: state_to_record}
+        return {
+            self.namespace_variable: state_to_record,
+            'rollback_vars': {
+                'vms': []
+            }
+        }
+
+
+class CheckPointVm(action.Action):
+    def __init__(self, init,
+                 var_info="info"):
+        self.var_info = var_info
+        super(CheckPointVm, self).__init__(init)
+
+    def run(self, rollback_vars=None, *args, **kwargs):
+        info = kwargs[self.var_info]['instances']
+        vm = info[info.keys()[0]]
+        rollback_vars_local = copy.deepcopy(rollback_vars)
+        success_vms = rollback_vars_local['vms']
+        pair_vm = {
+            'src_id': vm['old_id'] if 'old_id' in vm else '',
+            'dst_id': info.keys()[0]
```

Comment:
Ok, would be much more readable if `info.keys()[0]` would be replaced with `instance_id = info.keys()[0]`